playground/backend/internal/components/cache_component.go

```diff
@@ -37,73 +38,129 @@ func NewService(cache cache.Cache, db db.Database) *CacheComponent {
 
 // GetSdkCatalogFromCacheOrDatastore returns the sdk catalog from the cache
 // - If there is no sdk catalog in the cache, gets the sdk catalog from the Cloud Datastore and saves it to the cache
-func (cp *CacheComponent) GetSdkCatalogFromCacheOrDatastore(ctx context.Context) ([]*entity.SDKEntity, error) {
-	sdks, err := cp.cache.GetSdkCatalog(ctx)
-	if err != nil {
-		logger.Errorf("error during getting the sdk catalog from the cache, err: %s", err.Error())
-		sdks, err = cp.db.GetSDKs(ctx)
+func (cp *CacheComponent) GetSdkCatalogFromCacheOrDatastore(ctx context.Context, cacheRequestTimeout time.Duration) ([]*entity.SDKEntity, error) {
```

Comment:
let's make `timeout` a parameter of a CacheComponent, what you think?

Reply:
It's not flexible to configure this parameter this way, i'd like to stay it as an env variable